ignition-server/main.go

```diff
@@ -28,7 +34,73 @@ var addr = "0.0.0.0:9090"
 // TODO (alberto): Support only https.
 // TODO (alberto): Support only token authenticated requests.
 // TODO (alberto): Metrics.
+
 func main() {
+	cmd := &cobra.Command{
+		Use: "ignition-server",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Help()
+			os.Exit(1)
+		},
+	}
+	cmd.AddCommand(NewStartCommand())
+
+	if err := cmd.Execute(); err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+		os.Exit(1)
+	}
+}
+
+type Options struct {
+	Namespace string
+	CertFile  string
+	KeyFile   string
+	TokenFile string
+}
+
+func NewStartCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Starts the ignition server",
+	}
+
+	opts := Options{
+		Namespace: "",
+		CertFile:  "/var/run/secrets/ignition/tls.crt",
+		KeyFile:   "/var/run/secrets/ignition/tls.key",
+		TokenFile: "/var/run/secrets/ignition/token",
+	}
+
+	cmd.Flags().StringVar(&opts.Namespace, "namespace", "", "The namespace this server lives in")
+	cmd.Flags().StringVar(&opts.CertFile, "cert-file", opts.CertFile, "Path to the serving cert")
+	cmd.Flags().StringVar(&opts.KeyFile, "key-file", opts.KeyFile, "Path to the serving key")
+	cmd.Flags().StringVar(&opts.TokenFile, "token-file", opts.TokenFile, "Path to the auth token")
+	cmd.MarkFlagRequired("namespace")
```

Comment:
where is this used?

Reply:
Leftovers from a discarded experiment, deleted. Thanks